tests/chaos/oadp/conftest.py

```diff
@@ -11,29 +13,38 @@
     TIMEOUT_10MIN,
 )
 from utilities.infra import ExecCommandOnPod, wait_for_node_status
-from utilities.oadp import VeleroBackup, create_rhel_vm
-from utilities.storage import write_file
-from utilities.virt import node_mgmt_console, wait_for_node_schedulable_status
+from utilities.oadp import VeleroBackup
+from utilities.storage import data_volume_template_with_source_ref_dict, write_file
+from utilities.virt import VirtualMachineForTests, node_mgmt_console, wait_for_node_schedulable_status
 
 LOGGER = logging.getLogger(__name__)
 
 
-@pytest.fixture()
-def rhel_vm_with_dv_running(request, admin_client, chaos_namespace, snapshot_storage_class_name_scope_module):
+@pytest.fixture(scope="class")
+def rhel_vm_with_dv_running(admin_client, chaos_namespace, snapshot_storage_class_name_scope_module):
     """
-    Create a RHEL VM with a DataVolume.
+    Create a RHEL VM with a DataVolume for the whole test class.
     """
-    vm_name = request.param["vm_name"]
 
-    with create_rhel_vm(
-        storage_class=snapshot_storage_class_name_scope_module,
-        namespace=chaos_namespace.name,
-        vm_name=vm_name,
-        dv_name=f"dv-{vm_name}",
+    vm_name = "vm-oadp-chaos"
+    with VirtualMachineForTests(
         client=admin_client,
-        wait_running=True,
-        rhel_image=request.param["rhel_image"],
+        name=vm_name,
+        namespace=chaos_namespace.name,
+        vm_instance_type_infer=True,
+        vm_preference_infer=True,
+        data_volume_template=data_volume_template_with_source_ref_dict(
+            data_source=DataSource(
+                name=py_config["latest_rhel_os_dict"]["data_source"],
+                namespace=py_config.get(
+                    "golden_images_namespace_name",
+                    "openshift-virtualization-os-images",
+                ),
+            ),
+            storage_class=snapshot_storage_class_name_scope_module,
+        ),
     ) as vm:
+        vm.start(wait=True)
```

Comment:
`write_file()` does this start if vm is not ready.
Should `running_vm()` be used here to ensure we are ready to connect and write the file?

Reply:
Updated `write_file()` to make sure that VM is truly running.